CircleSpawn/CircleSpawn/ViewControllers/CircleSpawnController.swift

```diff
@@ -0,0 +1,116 @@
+import UIKit
+
+class CircleSpawnController: UIViewController, UIGestureRecognizerDelegate {
+
+    override func loadView() {
+        view = UIView()
+        view.backgroundColor = .white
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupGR()
+    }
+
+    private func setupGR() {
+        let doubleTap = UITapGestureRecognizer(target: self, action: #selector(onDoubleTap(_:)))
+        doubleTap.numberOfTapsRequired = 2
+        view.addGestureRecognizer(doubleTap)
+    }
+
+    @objc func onDoubleTap(_ doubleTap: UITapGestureRecognizer) {
+        let circleView = createCircleView(doubleTap)
+
+        let tripleTap = UITapGestureRecognizer(target: self, action: #selector(onTripleTap(_:)))
+        tripleTap.numberOfTapsRequired = 3
+        circleView.addGestureRecognizer(tripleTap)
+
+        doubleTap.require(toFail: tripleTap)
+
+        let longTap = UILongPressGestureRecognizer(target: self, action: #selector(onLongTap(_:)))
+        circleView.addGestureRecognizer(longTap)
+    }
+
+    private func createCircleView(_ doubleTap: UITapGestureRecognizer) -> CircleView {
+        let circleView = CircleView()
+        circleView.center = doubleTap.location(in: view)
+        circleView.alpha = 0.0
+        circleView.transform = CGAffineTransform(scaleX: 2.0, y: 2.0)
+        view.addSubview(circleView)
+
+        UIView.animate(withDuration: 0.2, delay: 0.1, usingSpringWithDamping: 0.5, initialSpringVelocity: 2, animations: {
+            circleView.alpha = 1.0
+            circleView.transform = .identity
+        })
+
+        return circleView
+    }
+
+    @objc func onTripleTap(_ tripleTap: UITapGestureRecognizer) {
+        guard let view = tripleTap.view else { return }
+
+        UIView.animate(withDuration: 0.2, animations: {
+            view.alpha = 0.0
+            view.transform = CGAffineTransform(scaleX: 2.0, y: 2.0)
+        }) { _ in
+            view.removeFromSuperview()
+        }
+    }
+
+    private var beginTapLocations: [UIView: CGPoint] = [:]
+
+    @objc func onLongTap(_ longTap: UITapGestureRecognizer) {
```

Comment:
Ajj... To nie jest `UITapGestureRecognizer`, tylko `UILongPressGestureRecognizer`. To w ogóle działa i nie crashuje?

Reply:
Ojej, nie zauważyłam :| 
To działa bo używałam tylko longTap.view i longTap.location, a to jest z UIGestureRecognizer